apps/client/src/components/PermissionForm.vue

```
@@ -40,31 +40,43 @@ const setPermissions = () => {
 }
 
 const addPermission = async (userEmail) => {
-  const userId = usersToLicence.value.find(user => user.email === userEmail).id
-  try {
-    await projectPermissionStore.addPermission(environment.value.id, { userId, level: 0 })
-  } catch (error) {
-    snackbarStore.setMessage(error?.message, 'error')
+  if (!project.value.locked) {
+    const userId = usersToLicence.value.find(user => user.email === userEmail).id
+    try {
+      await projectPermissionStore.addPermission(environment.value.id, { userId, level: 0 })
+    } catch (error) {
+      snackbarStore.setMessage(error?.message, 'error')
+    }
```

Je ne sais pas si ça a un intérêt mais on pourrait faire un `fail fast` qui affiche un message dans la snackbar si le `project.value.locked est true`

La question peut aussi se poser sur d'autres bout de code, je ne sais pas trop ce que tu en penses

---

le if est déjà une sécurité de plus vu que le bouton est `disabled` mais je peux rajouter le catch